src/Module/RemoteFollow.php

```diff
@@ -0,0 +1,105 @@
+<?php
+
+namespace Friendica\Module;
+
+use Friendica\BaseModule;
+use Friendica\DI;
+use Friendica\Core\Logger;
+use Friendica\Core\Protocol;
+use Friendica\Core\Renderer;
+use Friendica\Core\Search;
+use Friendica\Core\System;
+use Friendica\Model\Profile;
+use Friendica\Network\Probe;
+
+/**
+ * Remotely follow the account on this system by the provided account
+ */
+class RemoteFollow extends BaseModule
+{
+	public static function init(array $parameters = [])
+	{
+		if (empty($parameters['profile'])) {
+			return;
+		}
+
+		Profile::load(DI::app(), $parameters['profile']);
+	}
+
+	public static function post(array $parameters = [])
+	{
+		$a = DI::app();
+
+		if (empty($parameters['profile']) || !empty($_POST['cancel']) || empty($_POST['dfrn_url'])) {
+			DI::baseUrl()->redirect();
+		}
+	
+		if (empty($a->profile['uid'])) {
+			notice(DI::l10n()->t('Profile unavailable.') . EOL);
+			return;
+		}
+		
+		$url = trim($_POST['dfrn_url']);
+		if (!strlen($url)) {
+			notice(DI::l10n()->t("Invalid locator") . EOL);
+			return;
+		}
+
+		// Detect the network, make sure the provided URL is valid
+		$data = Probe::uri($url);
+		if ($data['network'] == Protocol::PHANTOM) {
+			notice(DI::l10n()->t("The provided profile link doesn't seem to be valid"));
+			return;
+		}
+
+		// Fetch link for the "remote follow" functionality of the given profile
+		$follow_link_template = Probe::getRemoteFollowLink($url);
+
+		if (empty($follow_link_template)) {
+			notice(DI::l10n()->t("Remote subscription can't be done for your network. Please subscribe directly on your system."));
+			return;
+		}
+
+		Logger::notice('Remote request', ['url' => $url, 'follow' => $a->profile['url'], 'remote' => $follow_link_template]);
+		
+		// Substitute our user's feed URL into $follow_link_template
+		// Send the subscriber home to subscribe
+		// Diaspora needs the uri in the format user@domain.tld
+		if ($data['network'] == Protocol::DIASPORA) {
+			$uri = urlencode($a->profile['addr']);
+		} else {
+			$uri = urlencode($a->profile['url']);
+		}
+	
+		$follow_link = str_replace('{uri}', $uri, $follow_link_template);
+		System::externalRedirect($follow_link);
+	}
+
+	public static function content(array $parameters = [])
+	{
+		$a = DI::app();
+
+		if (empty($parameters['profile']) || empty($a->profile)) {
```

Comment:
You don't have to check for the existence/emptiness of `$parameters['profile']`, the router will not select this module for empty values of `{profile}` in the route.

Reply:
Okay.